File: SOTALog/SOTALog/Models/POTAPark.swift
import Foundation
import GRDB

struct POTAPark: Codable, Identifiable, Equatable {
    var reference: String
    var name: String
    var referenceNormalized: String?
    var latitude: Double?
    var longitude: Double?
    var locationDesc: String?

    var id: String { reference }

    /// Display string: "US-4431 Prescott NF"
    var displayName: String {
        "\(reference) \(name)"
    }

    /// Strips dashes and uppercases: "US-4431" → "US4431"
    static func normalize(_ reference: String) -> String {
        reference.replacingOccurrences(of: "-", with: "").uppercased()
    }
}

extension POTAPark: FetchableRecord, PersistableRecord {
    static var databaseTableName = "potaPark"
}

File: android/app/src/main/kotlin/com/sotalog/android/domain/models/POTAPark.kt
package com.sotalog.android.domain.models

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "potaPark")
data class POTAPark(
    @PrimaryKey
    val reference: String,
    val name: String,
    val referenceNormalized: String? = null,
    val latitude: Double? = null,
    val longitude: Double? = null,
    val locationDesc: String? = null,
) {
    val displayName: String get() = "$reference $name"

    companion object {
        fun normalize(reference: String): String =
            reference.replace("-", "").uppercase()
    }
}
